### PR TITLE
[Feature] Indicator Metrics

### PR DESCRIPTION
**Pull request by gertie-sheshe**
## Description

Add Indicator Metrics Component

Fixes # (issue)

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## Screenshots
Screenshot 2021-09-13 at 13 15 23

## Checklist:

- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation

### CODE REVIEW

**Review by KobbyMmo — COMMENTED** [review]
looks good @gertie-sheshe

**Review by kilemensi — COMMENTED** [review]
👍🏾 @gertie-sheshe 

-----------

It's @KhadijaMahanga's call but since we're building a UI _for_ Wazimap API, my suggestion would be we should use the exact or very similar naming scheme. For example Wazimap calls this component [`ProfileKeyMetric`](https://openup.gitbook.io/wazi-ng-technical/system-architecture/database-models). I think we should call it the same (this extends to "props" as well)

**Comment by KhadijaMahanga** [comment]
> nce we're building a UI _for_ Wazimap API, my suggest

@gertie-sheshe we can rename this component to `ProfileKeyMetric`

File: src/components/HURUmap/Metrics/index.js [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+  value,
+}) => {
+  const classes = useStyles();
+
+  return (
+    <>
+      <div className={classes.root}>
+        <Typography variant="h3">{percentage}</Typography>
+        <div className={classes.description}>
+          <Typography variant="caption">{description}</Typography>
```
[the remark this answers is not in the document]
Comment (gertie-sheshe) [reply src/components/HURUmap/Metrics/index.js]: Fixed

File: src/components/HURUmap/Metrics/index.js [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+import PropTypes from "prop-types";
+import React from "react";
+
+import useStyles from "./useStyles";
+
+const Metrics = ({
+  percentage,
+  description,
+  color,
+  bottomDescription,
```
[the remark this answers is not in the document]
Comment (gertie-sheshe) [reply src/components/HURUmap/Metrics/index.js]: I agree. Fixed

File: src/components/HURUmap/Metrics/index.js [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+}) => {
+  const classes = useStyles();
+
+  return (
+    <>
+      <div className={classes.root}>
+        <Typography variant="h3">{percentage}</Typography>
+        <div className={classes.description}>
+          <Typography variant="caption">{description}</Typography>
+        </div>
```
[the remark this answers is not in the document]
Comment (gertie-sheshe) [reply src/components/HURUmap/Metrics/index.js]: It needed a block element to pick margin styling. Removed the div and set the display to `block`. Fixed

File: src/components/HURUmap/Metrics/index.js [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+        <LinearProgress
+          className={classes.progressBar}
+          value={value}
+          color={color}
+          variant="buffer"
+        />
+      </div>
+      <Typography variant="caption" className={classes.bottomDescription}>
+        {bottomDescription}
+      </Typography>
```
[the remark this answers is not in the document]
Comment (gertie-sheshe) [reply src/components/HURUmap/Metrics/index.js]: Correct. Fixed

File: src/components/HURUmap/Metrics/index.js [hunk 5]
```diff
@@ -0,0 +1,53 @@
...
+import React from "react";
+
+import useStyles from "./useStyles";
+
+const Metrics = ({
+  percentage,
+  description,
+  color,
+  bottomDescription,
+  value,
```
[the remark this answers is not in the document]
Comment (gertie-sheshe) [reply src/components/HURUmap/Metrics/index.js]: Fixed

File: src/components/HURUmap/Metrics/useStyles.js [hunk 6]
```diff
@@ -0,0 +1,25 @@
+import { makeStyles } from "@material-ui/core/styles";
+
+const useStyles = makeStyles(({ typography, palette }) => ({
+  root: {
+    backgroundColor: palette.background.paper,
+    maxWidth: typography.pxToRem(374),
```
[the remark this answers is not in the document]
Comment (gertie-sheshe) [reply src/components/HURUmap/Metrics/useStyles.js:6]: @kilemensi 🤔 
Screenshot 2021-09-14 at 11 26 00

**Review by nerdlet — APPROVED** [review]
@gertie-sheshe Looks good 🚀

File: src/components/HURUmap/Metrics/useStyles.js (hunk 6 above)
[the remark this answers is not in the document]
Comment (kilemensi) [reply src/components/HURUmap/Metrics/useStyles.js:6]: Could it be we're looking at different pages @gertie-sheshe ... I'm looking at [this](https://xd.adobe.com/view/5aa579ae-4704-4079-abe2-e429d3e5e54f-0ff6/screen/86e9de1b-8ec5-4868-a12e-bc76c6f908a3/specs/) one.

Screenshot from 2021-09-14 13-38-39

[the remark this answers is not in the document]
Comment (gertie-sheshe) [reply src/components/HURUmap/Metrics/useStyles.js:6]: Ah yeah, that is it. [This](https://xd.adobe.com/view/5aa579ae-4704-4079-abe2-e429d3e5e54f-0ff6/screen/8e082cd7-9b67-4589-a967-74ca97bbe175) is the one I am looking at. @kilemensi

File: src/components/HURUmap/Metrics/index.stories.js [hunk 7]
```diff
@@ -0,0 +1,19 @@
...
+  title: "Components/HURUmap/Metrics",
+  argTypes: {},
+};
+
+const Template = ({ ...args }) => <Metrics {...args} />;
+
+export const Default = Template.bind({});
+
+Default.args = {
+  color: "Primary",
```
Comment (KhadijaMahanga) [inline src/components/HURUmap/Metrics/index.stories.js:15]: @gertie-sheshe what other color options do we have? Tried to change color on storybook and color doesn't change

File: src/components/HURUmap/Metrics/index.js [hunk 8]
```diff
@@ -0,0 +1,50 @@
...
+
+import useStyles from "./useStyles";
+
+const Metrics = ({ percentage, description, color, summary }) => {
+  const classes = useStyles();
+
+  return (
+    <>
+      <div className={classes.root}>
+        <Typography variant="h3">{`${percentage}%`}</Typography>
```
Comment (KhadijaMahanga) [inline src/components/HURUmap/Metrics/index.js:13]: Metrics won't always be `percent values`.  `%`

File: src/components/HURUmap/Metrics/index.js [hunk 9]
```diff
@@ -0,0 +1,50 @@
...
+        <LinearProgress
+          className={classes.progressBar}
+          value={percentage}
+          color={color}
+          variant="buffer"
+        />
+      </div>
+      <Typography
+        variant="caption"
+        className={`${classes.summary} ${classes.text}`}
```
Comment (KhadijaMahanga) [inline src/components/HURUmap/Metrics/index.js:28]: `clsx`, and lets check if summary is defined

File: src/components/HURUmap/Metrics/index.stories.js (hunk 7 above)
Comment (gertie-sheshe) [reply src/components/HURUmap/Metrics/index.stories.js]: There are [two](https://material-ui.com/api/linear-progress/): `Primary` and `Secondary @KhadijaMahanga Both match the colors used in the design

File: src/components/HURUmap/Metrics/index.js (hunk 8 above)
Comment (gertie-sheshe) [reply src/components/HURUmap/Metrics/index.js]: Oh..what are the other options so I can know what to show @KhadijaMahanga

File: src/components/HURUmap/Metrics/index.js (hunk 9 above)
Comment (gertie-sheshe) [reply src/components/HURUmap/Metrics/index.js]: Fixed

File: src/components/HURUmap/ProfileKeyMetric/index.js [hunk 10]
```diff
@@ -0,0 +1,58 @@
+import { Typography, LinearProgress } from "@material-ui/core";
+import clsx from "clsx";
+import PropTypes from "prop-types";
+import React from "react";
+
+import useStyles from "./useStyles";
+
+const ProfileKeyMetric = ({ percentage, description, color, summary }) => {
```
Comment (kilemensi) [inline src/components/HURUmap/ProfileKeyMetric/index.js:8]:
```suggestion
const KeyMetric = ({ formatedValue, title, color, description }) => {
```

As we discussed in the call:

1. `KeyMetric` is part of `SubCategory` and not `Profile`.
2. We should default to using `title`, `value`, and `description` in all our components,
3. `formatedValue` if available should be used when we need to show `value` with formatting e.g. percentage, currency, etc.

File: src/components/HURUmap/ProfileKeyMetric/index.js [hunk 11]
```diff
@@ -0,0 +1,58 @@
...
+  const classes = useStyles();
+
+  return (
```
Comment (kilemensi) [inline src/components/HURUmap/ProfileKeyMetric/index.js:11]:
```suggestion
  if (!(value && title)) {
    return null;
  }
  return (
```

File: src/components/HURUmap/ProfileKeyMetric/index.js [hunk 12]
```diff
@@ -0,0 +1,58 @@
...
+            value={percentage}
+            color={color}
+            variant="buffer"
+          />
+        )}
+      </div>
+      {summary && (
+        <Typography
+          variant="caption"
+          className={clsx(classes.summary, classes.text)}
```
Comment (kilemensi) [inline src/components/HURUmap/ProfileKeyMetric/index.js:35]:
```suggestion
          className={clsx(classes.text, classes.description)}
```

**Review by kilemensi — COMMENTED** [review]
👍🏾

File: src/components/HURUmap/ProfileKeyMetric/index.js (hunk 11 above)
Comment (gertie-sheshe) [reply src/components/HURUmap/ProfileKeyMetric/index.js]: Fixed

File: src/components/HURUmap/ProfileKeyMetric/index.js (hunk 12 above)
Comment (gertie-sheshe) [reply src/components/HURUmap/ProfileKeyMetric/index.js]: Fixed

File: src/components/HURUmap/ProfileKeyMetric/index.js (hunk 10 above)
Comment (gertie-sheshe) [reply src/components/HURUmap/ProfileKeyMetric/index.js]: Quick question on the `formattedValues` - Do we get both `formattedValue` and `value` and does the formattedValue come with it's sign i.e % or £ etc? @kilemensi

Comment (kilemensi) [reply src/components/HURUmap/ProfileKeyMetric/index.js]: Yes and Yes @gertie-sheshe

Comment (gertie-sheshe) [reply src/components/HURUmap/ProfileKeyMetric/index.js]: Fixed

**Review by KobbyMmo — APPROVED** [review]
looks good to me

File: src/components/HURUmap/KeyMetric/index.js [hunk 13]
```diff
@@ -0,0 +1,58 @@
...
+import clsx from "clsx";
+import PropTypes from "prop-types";
+import React from "react";
+
+import useStyles from "./useStyles";
+
+const KeyMetric = ({ formattedValue, value, title, color, description }) => {
+  const classes = useStyles();
+
+  if (!(formattedValue && title)) {
```
Comment (kilemensi) [inline src/components/HURUmap/KeyMetric/index.js:11]:
```suggestion
  if (!(value && title)) {
```

`formattedValue` is option; we use it when available

File: src/components/HURUmap/KeyMetric/index.js [hunk 14]
```diff
@@ -0,0 +1,58 @@
...
+          color={color}
+          variant="buffer"
+        />
+      </div>
+      <Typography
+        variant="caption"
+        className={clsx(classes.description, classes.text)}
+      >
+        {description}
+      </Typography>
```
Comment (kilemensi) [inline src/components/HURUmap/KeyMetric/index.js:37]:
```suggestion
      {description ? (
        <Typography
          variant="caption"
          className={clsx(classes.text, classes.description)}
        >
          {description}
        </Typography>
      ) : null}
```
description is also optional.

**Review by kilemensi — APPROVED** [review]
🚀

File: src/components/HURUmap/KeyMetric/index.js (hunk 14 above)
Comment (gertie-sheshe) [reply src/components/HURUmap/KeyMetric/index.js]: Fixed

File: src/components/HURUmap/KeyMetric/index.js (hunk 13 above)
Comment (gertie-sheshe) [reply src/components/HURUmap/KeyMetric/index.js]: Got it. Fixed

**Pull request merged**